pkg/k8s/watchers/cilium_node.go

```diff
@@ -36,9 +36,6 @@ func (k *K8sWatcher) ciliumNodeInit(ciliumNPClient *k8s.K8sCiliumClient, asyncCo
 					if ciliumNode := k8s.ObjToCiliumNode(obj); ciliumNode != nil {
 						valid = true
 						n := nodeTypes.ParseCiliumNode(ciliumNode)
-						if n.IsLocal() {
-							return
-						}
```

Comment:
I don't know the exact reason here but I _suspect_ that the bug I proposed to fix in https://github.com/cilium/cilium/pull/17836 was causing the identity to be pushed down incorrectly during the initial, local, NodeDiscovery path for this.. so then to fix this @christarazi had moved the checks  here into the Update function and then ensured that the `IdentityMetadata` was updated from this path and that it would trigger the label evaluation.

I can try reinstating these changes and see whether things work from there when including the above PR 17836.